front/components/data_source/ConnectorPermissionsModal.tsx

```diff
@@ -618,6 +618,13 @@ export function ConnectorPermissionsModal({
     dataSource.connectorProvider === "notion" &&
     featureFlags.includes("advanced_notion_management");
 
+  const getNodeParents = (node: ContentNodeWithParent) => {
+    if (node.parentInternalId) {
```

Comment:
I'm not familiar with this code. Nodes can have both a parentInternalId (singular) and parentInternalIds (plurial)? Are they expected to be consistent, assuming they're both set? Or are they never both set?

The change is probably fine, but I can't give LGTM with great confidence as I'm a bit confused.